### PR TITLE
restore setuid feature with using SYS_SETREUID

### PR DESCRIPTION
**Pull request by kovetskiy**
Now it uses SYS_SETREUID for properly restoring effective uid after
checkSetuid, also added manual restoring original effective uid.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -122,6 +142,34 @@ func main() {
 	}
 }
 
+// checkSeteuid calls syscall SYS_SETREUID in additional goroutine with new uid
+// and tries to restore original euid
+func checkSeteuid(uid int) (err error) {
+	waiting := sync.WaitGroup{}
+	waiting.Add(1)
+
+	go func() {
```
Comment (seletskiy) [inline main.go]: No need in go-routine now, because UID got restored.

Comment (kovetskiy) [inline main.go]: @seletskiy fixed up with force push

**Comment by kovetskiy** [comment]
@seletskiy fixed up with force push

**Comment by seletskiy** [comment]
@zazab: please, revert #6.

**Pull request merged**